river/PointerMapping.zig

```diff
@@ -14,13 +14,47 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
+const Self = @This();
+
 const wlr = @import("wlroots");
 
+const util = @import("util.zig");
+
 pub const Action = enum {
     move,
     resize,
+    command,
 };
 
 event_code: u32,
 modifiers: wlr.Keyboard.ModifierMask,
 action: Action,
+command_args: []const [:0]const u8,
+
+pub fn init(
+    event_code: u32,
+    modifiers: wlr.Keyboard.ModifierMask,
+    action: Action,
+    command_args: []const [:0]const u8,
+) !Self {
+    const args_len = if (action == Action.command) command_args.len else 0;
```

Comment:
I'd prefer a cleaner separation of the command and non-command cases.

Reply:
Refactored as switch.